From 1f5d8228ca2c06ee1baa81b40cdba824ed768d95 Mon Sep 17 00:00:00 2001
From: Arnd Bergmann <arnd@arndb.de>
Date: Mon, 10 Aug 2026 16:59:07 +0200
Subject: [PATCH 12822/15333] Revert "riscv: dts: spacemit: k3: add i2s0-i2s5
 nodes"

This reverts commit cd01c677869affe1b01403ca2fad8fc0fe85e8bd, which
caused a build failure:

Lexical error: ../arch/riscv/boot/dts/spacemit/k3.dtsi:1181.20-39 Unexpected 'CLK_MPMU_I2S_SYSCLK'

Signed-off-by: Arnd Bergmann <arnd@arndb.de>
---
 arch/riscv/boot/dts/spacemit/k3.dtsi | 119 ---------------------------
 1 file changed, 119 deletions(-)

diff --git a/arch/riscv/boot/dts/spacemit/k3.dtsi b/arch/riscv/boot/dts/spacemit/k3.dtsi
index 0eb24fac4733..c3f2dce0969c 100644
--- a/arch/riscv/boot/dts/spacemit/k3.dtsi
+++ b/arch/riscv/boot/dts/spacemit/k3.dtsi
@@ -1170,125 +1170,6 @@ spi1: spi@d4040800 {
 			status = "disabled";
 		};
 
-		i2s0: i2s@d4026000 {
-			compatible = "spacemit,k3-i2s";
-			reg = <0x0 0xd4026000 0x0 0x30>;
-			clocks = <&syscon_mpmu CLK_MPMU_I2S0_SYSCLK>,
-				 <&syscon_apbc CLK_APBC_I2S0_BCLK>,
-				 <&syscon_apbc CLK_APBC_I2S0_BUS>,
-				 <&syscon_apbc CLK_APBC_I2S0>,
-				 <&syscon_mpmu CLK_MPMU_I2S0_SYSCLK_DIV>,
-				 <&syscon_mpmu CLK_MPMU_I2S_SYSCLK>,
-				 <&syscon_mpmu CLK_MPMU_I2S_BCLK>;
-			clock-names = "sysclk", "bclk", "bus", "func",
-				      "sysclk_div", "c_sysclk", "c_bclk";
-			dmas = <&pdma 21>, <&pdma 22>;
-			dma-names = "tx", "rx";
-			resets = <&syscon_apbc RESET_APBC_I2S0>;
-			#sound-dai-cells = <0>;
-
-			status = "disabled";
-		};
-
-		i2s1: i2s@d4026800 {
-			compatible = "spacemit,k3-i2s";
-			reg = <0x0 0xd4026800 0x0 0x30>;
-			clocks = <&syscon_mpmu CLK_MPMU_I2S1_SYSCLK>,
-				 <&syscon_apbc CLK_APBC_I2S1_BCLK>,
-				 <&syscon_apbc CLK_APBC_I2S1_BUS>,
-				 <&syscon_apbc CLK_APBC_I2S1>,
-				 <&syscon_mpmu CLK_MPMU_I2S_SYSCLK>,
-				 <&syscon_mpmu CLK_MPMU_I2S_BCLK>;
-			clock-names = "sysclk", "bclk", "bus", "func",
-				      "c_sysclk", "c_bclk";
-			dmas = <&pdma 23>, <&pdma 24>;
-			dma-names = "tx", "rx";
-			resets = <&syscon_apbc RESET_APBC_I2S1>;
-			#sound-dai-cells = <0>;
-
-			status = "disabled";
-		};
-
-		i2s2: i2s@d4027000 {
-			compatible = "spacemit,k3-i2s";
-			reg = <0x0 0xd4027000 0x0 0x30>;
-			clocks = <&syscon_mpmu CLK_MPMU_I2S2_SYSCLK>,
-				 <&syscon_apbc CLK_APBC_I2S2_BCLK>,
-				 <&syscon_apbc CLK_APBC_I2S2_BUS>,
-				 <&syscon_apbc CLK_APBC_I2S2>,
-				 <&syscon_mpmu CLK_MPMU_I2S2_SYSCLK_DIV>,
-				 <&syscon_mpmu CLK_MPMU_I2S_SYSCLK>,
-				 <&syscon_mpmu CLK_MPMU_I2S_BCLK>;
-			clock-names = "sysclk", "bclk", "bus", "func",
-				      "sysclk_div", "c_sysclk", "c_bclk";
-			dmas = <&pdma 56>, <&pdma 57>;
-			dma-names = "tx", "rx";
-			resets = <&syscon_apbc RESET_APBC_I2S2>;
-			#sound-dai-cells = <0>;
-
-			status = "disabled";
-		};
-
-		i2s3: i2s@d4027800 {
-			compatible = "spacemit,k3-i2s";
-			reg = <0x0 0xd4027800 0x0 0x30>;
-			clocks = <&syscon_mpmu CLK_MPMU_I2S3_SYSCLK>,
-				 <&syscon_apbc CLK_APBC_I2S3_BCLK>,
-				 <&syscon_apbc CLK_APBC_I2S3_BUS>,
-				 <&syscon_apbc CLK_APBC_I2S3>,
-				 <&syscon_mpmu CLK_MPMU_I2S3_SYSCLK_DIV>,
-				 <&syscon_mpmu CLK_MPMU_I2S_SYSCLK>,
-				 <&syscon_mpmu CLK_MPMU_I2S_BCLK>;
-			clock-names = "sysclk", "bclk", "bus", "func",
-				      "sysclk_div", "c_sysclk", "c_bclk";
-			dmas = <&pdma 58>, <&pdma 59>;
-			dma-names = "tx", "rx";
-			resets = <&syscon_apbc RESET_APBC_I2S3>;
-			#sound-dai-cells = <0>;
-
-			status = "disabled";
-		};
-
-		i2s4: i2s@d4041000 {
-			compatible = "spacemit,k3-i2s";
-			reg = <0x0 0xd4041000 0x0 0x30>;
-			clocks = <&syscon_mpmu CLK_MPMU_I2S4_SYSCLK>,
-				 <&syscon_apbc CLK_APBC_I2S4_BCLK>,
-				 <&syscon_apbc CLK_APBC_I2S4_BUS>,
-				 <&syscon_apbc CLK_APBC_I2S4>,
-				 <&syscon_mpmu CLK_MPMU_I2S4_SYSCLK_DIV>,
-				 <&syscon_mpmu CLK_MPMU_I2S_SYSCLK>,
-				 <&syscon_mpmu CLK_MPMU_I2S_BCLK>;
-			clock-names = "sysclk", "bclk", "bus", "func",
-				      "sysclk_div", "c_sysclk", "c_bclk";
-			dmas = <&pdma 60>, <&pdma 61>;
-			dma-names = "tx", "rx";
-			resets = <&syscon_apbc RESET_APBC_I2S4>;
-			#sound-dai-cells = <0>;
-
-			status = "disabled";
-		};
-
-		i2s5: i2s@d4041800 {
-			compatible = "spacemit,k3-i2s";
-			reg = <0x0 0xd4041800 0x0 0x30>;
-			clocks = <&syscon_mpmu CLK_MPMU_I2S5_SYSCLK>,
-				 <&syscon_apbc CLK_APBC_I2S5_BCLK>,
-				 <&syscon_apbc CLK_APBC_I2S5_BUS>,
-				 <&syscon_apbc CLK_APBC_I2S5>,
-				 <&syscon_mpmu CLK_MPMU_I2S5_SYSCLK_DIV>,
-				 <&syscon_mpmu CLK_MPMU_I2S_SYSCLK>,
-				 <&syscon_mpmu CLK_MPMU_I2S_BCLK>;
-			clock-names = "sysclk", "bclk", "bus", "func",
-				      "sysclk_div", "c_sysclk", "c_bclk";
-			dmas = <&pdma 62>, <&pdma 63>;
-			dma-names = "tx", "rx";
-			resets = <&syscon_apbc RESET_APBC_I2S5>;
-			#sound-dai-cells = <0>;
-
-			status = "disabled";
-		};
-
 		syscon_mpmu: system-controller@d4050000 {
 			compatible = "spacemit,k3-syscon-mpmu";
 			reg = <0x0 0xd4050000 0x0 0x10000>;
-- 
2.55.0

